hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplitThread.java

```diff
@@ -131,11 +132,12 @@ public Thread newThread(Runnable r) {
   private void createSplitExcecutors() {
     final String n = Thread.currentThread().getName();
     int splitThreads = conf.getInt(SPLIT_THREADS, SPLIT_THREADS_DEFAULT);
+    final AtomicInteger splitThreadCounter = new AtomicInteger(0);
     this.splits =
         (ThreadPoolExecutor) Executors.newFixedThreadPool(splitThreads, new ThreadFactory() {
           @Override
```

Comment:
Only one place not sure: shouldn't the counter be placed in `ThreadFactory`?

Reply:
+1 on what Reid said.

Reply:
And same way below also.

Reply:
Thanks @Reidddddd and @anoopsjohn  for the quick review, have moved the counter local to  ThreadFactory in commit-2.